### PR TITLE
Modifying LER with or'd bit_value or bitset_value worked

### PR DESCRIPTION
**Pull request by descipher**
Tested on LPC1768 (MKS SGEN L V1.0) running bugfix-2.0.x
Tested both PWM pins P1.23 and P2.00.  
Ran 100,000+ latches  0 missed over a span of two hours.
Verification = logic analyzer data out to excel, mathematical count of input to output.

### CODE REVIEW

**Comment by p3p** [comment]
Have been looking at this today and can confirm this does look like it works but I don't have my original test case anymore .. so slightly worried that I'm just not reproducing the edge case bug that needed latching disabled. 

It may be I never used the latch bits correctly although I do have old versions of the code that looks to be correctly updating that register.

File: system/lpc176x/HardwarePWM.h [hunk 1]
```diff
@@ -138,11 +134,9 @@ class HardwarePWM {
 
   // update the match register for a channel and set the latch to update on next period
   static inline void set_match(const pin_t pin, const uint32_t value) {
-    //work around for bug if MR1 == MR0
+    //work around for bug if MRn == MR0
     *match_register_ptr(pin) = value == LPC_PWM1->MR0 ? value + 1 : value;
-    // tried to work around latch issue by always setting all bits, was unsuccessful
-    LPC_PWM1->LER = util::bit_value(pin_has_pwm(pin));
```
Comment (descipher) [inline system/lpc176x/HardwarePWM.h:144]: With no OR function at this point there would be inconsistent behaviour when more than one pwm was active.

Comment (p3p) [reply system/lpc176x/HardwarePWM.h:144]: Indeed, other than re-enabling latching that's the only change in this PR, I wouldn't put it past myself to make such a stupid mistake and spend weeks trying to debug it (as I did at the time), but the comment seems to imply that I got frustrated at some point and just set all latch bits on every match update and it didn't work around the issue so I think not ORing the current register value was an artefact of that debugging and leaving latching disabled.

I'l do some more tests tomorrow, I'm mostly worried about the client application setting the match value but it not updating the signal when being spammed with match updates. (I think that was the original issue, .. I really have left this too long)

Comment (descipher) [reply system/lpc176x/HardwarePWM.h:144]: Yes, been there too, with the bit function toolset this would be so easy to overlook. 
e.g. bitset vs bit value function template, one template applies OR and one does not.
+1 for more testing, quality = testing. Thanks for your work or this.

**Comment by descipher** [comment]
👍 Thanks

**Comment by p3p** [comment]
Will wait and see if the weird latching issue that I'm sure I wasn't imagining reappears ^^

**Comment by descipher** [comment]
🤞
